.evergreen/config_generator/components/cse/openssl.py

```diff
@@ -0,0 +1,101 @@
+from shrub.v3.evg_build_variant import BuildVariant
+from shrub.v3.evg_task import EvgTaskRef
+
+from config_generator.etc.compile import generate_compile_tasks
+from config_generator.etc.function import merge_defns
+
+from config_generator.etc.cse.compile import CompileCommon
+from config_generator.etc.cse.test import generate_test_tasks
+
+
+SSL = 'openssl'
+TAG = f'cse-matrix-{SSL}'
+
+
+# pylint: disable=line-too-long
+# fmt: off
+COMPILE_MATRIX = [
+    ('debian10',          'gcc',       None, ['cyrus']),
+    ('debian11',          'gcc',       None, ['cyrus']),
+    ('debian92',          'clang',     None, ['cyrus']),
+    ('debian92',          'gcc',       None, ['cyrus']),
+    ('rhel80',            'gcc',       None, ['cyrus']),
+    ('rhel83-zseries',    'gcc',       None, ['cyrus']),
+    ('ubuntu1604',        'clang',     None, ['cyrus']),
+    ('ubuntu1804-arm64',  'gcc',       None, ['cyrus']),
+    ('ubuntu1804',        'gcc',       None, ['cyrus']),
+    ('ubuntu2004',        'gcc',       None, ['cyrus']),
+    ('windows-64-vs2017', 'vs2017x64', None, ['cyrus']),
+]
+
+# TODO (CDRIVER-3789): test cse with the 'sharded' topology.
+TEST_MATRIX = [
+    ('rhel83-zseries',    'gcc',       None, 'cyrus', ['auth'], ['server'], ['4.2', '4.4', '5.0']),
```

Comment:
It seems like [4.2](https://spruce.mongodb.com/task/mongo_c_driver_cse_matrix_openssl_cse_sasl_cyrus_openssl_rhel83_zseries_gcc_test_4.2_server_auth_patch_8b5a14e5a35fb7b365a1a53f24c933f8c497f337_63e5517461837d4373055658_23_02_09_20_03_02/logs?execution=0) and [4.4](https://spruce.mongodb.com/task/mongo_c_driver_cse_matrix_openssl_cse_sasl_cyrus_openssl_rhel83_zseries_gcc_test_4.4_server_auth_patch_8b5a14e5a35fb7b365a1a53f24c933f8c497f337_63e5517461837d4373055658_23_02_09_20_03_02/logs?execution=0) are not available on `rhel83-zseries`. Shall we remove them from the matrix or apply an alternative solution?

Reply:
I think that we just remove them.  The server only supports 5.0 and newer on RHEL8 on zSeries.  To get 4.4/4.2 we have to go back to RHEL7, but my understanding is that those hosts are deprecated.  That said, using RHEL7 to get 4.4/4.2 test coverage seems like a reasonable thing and I don't think that it would violate any deprecation principle (being that 4.4/4.2 are legacy versions at this point).

Reply:
Given 4.2 and 4.4 tests are being covered by other distros below, and I do not expect there to be significant differences in code behavior across platforms, I think I will opt to remove them rather than adding deprecated distros.